cmake/Dependencies.cmake

```diff
@@ -318,7 +318,18 @@ include_directories(SYSTEM ${EIGEN3_INCLUDE_DIR})
 
 # ---[ Python + Numpy
 if(BUILD_PYTHON)
-  set(Python_ADDITIONAL_VERSIONS 3.6 3.5 2.8 2.7 2.6)
+  # Put the currently-activated python version at the front of
+  # Python_ADDITIONAL_VERSIONS so that it is found first. Otherwise there may
+  # be mismatches between Python executables and libraries used at different
+  # stages of build time and at runtime
```

Comment:
This seems like an oddly oblique way to get what you want. Given that you want to pin to whatever `python` is (BTW, not *necessarily* always the case), why don't you just invoke Python to get all of the information you need?

The scheme as it is here, couldn't I hypothetically get a Python that is different from `python`, but "happens" to have the same version?

Reply:
It seems like a proper fix would be to have find_package(PythonInterp/PythonLibs) merged or have them handle finding the same Python. However those changes might be a bit overkill, as they'll only be needed when people have complicated python setups with multiple installations of the same version. If people complain about problems with picking up different python libs/executable even with these changes, then I'd rather help them fix their python setups than change this logic.

These changes are simple and fix an error that can occur on a clean anaconda install, so I will check these in.

Reply:
A little bit of extra sanity checking (e.g., checking that the second Python executable you found is, in fact, the one that is in the PATH), in this case, will go a long way in debugging these issues in the wild.